samples/WOCCatalog/WOCCatalog/URLSessionViewController.mm

```diff
@@ -246,8 +261,12 @@ - (void)downloadTaskButtonPressed:(id)sender {
 - (void)cancelLastButtonPressed:(id)sender {
     if ([_lastTask isKindOfClass:[NSURLSessionDownloadTask class]]) {
         [(NSURLSessionDownloadTask*)_lastTask cancelByProducingResumeData:^(NSData* resumeData) {
-            [self _printOutput:@"Download task produced resume data (size %d)\n", resumeData.length];
-            _lastResumeData = resumeData;
+            if (resumeData == nil) {
+                [self _printOutput:@"Download task was cancelled but cannot be resumed"];
+            } else {
+                [self _printOutput:@"Download task was cancelled and produced resume data (size %d)\n", resumeData.length];
+                _lastResumeData = resumeData;
```

Comment:
>stResumeData = resumeData; [](start = 19, length = 26)

Since we only use one var for the last resume data, we should clear it if we didn't get one in this callback. Otherwise, it could try to resume a different cancelled download.